derive-input/src/internal/dsl/method.rs

```diff
@@ -1625,6 +1679,19 @@ pub(in crate::internal) fn for_method(
                                 };
                             };
 
+                            let hook_call = if spacetimedsl_table.hook_on_delete {
+                                let hook_fn_name = format_ident!("on_{}_delete", singular_table_name);
+                                quote! {
+                                    if let Some(deleted_row) = &row_to_delete {
+                                        if let Err(error_msg) = #hook_fn_name(self, deleted_row) {
+                                            return Err(spacetimedsl::SpacetimeDSLError::Error(error_msg));
+                                        }
+                                    }
```

Comment:
hook functions should return `spacetimedsl::SpacetimeDSLError` instead of string.